[INTERNAL] Add CircleCI config

elskwid opened the pull request:
~~# ⚠️ Requires #4 to be merged first~~

# Overview 

Adds CircleCI config and enables builds on circleci.com.

<img width="1395" alt="wait-for-netlify-action 2022-02-09 00-26-06" src="https://user-images.githubusercontent.com/9002/153155399-e2e9df0d-b62b-416b-9345-b9e31a07bdca.png">

Commit:
Add CircleCI config

phlipper commented on .circleci/config.yml:
```diff
@@ -0,0 +1,41 @@
+executors:
+  node-docker:
+    docker:
+      - image: mutations/node:16.13.0-alpine3.14-2021-11-04
```
This should be updated to `mutations/node:16.13.2-alpine3.15-2022-01-18` to match the version in the [`docker-compose.yml`](https://github.com/mutations/wait-for-netlify-action/pull/2/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R15) from #2.

```suggestion
      - image: mutations/node:16.13.2-alpine3.15-2022-01-18
```

phlipper reviewed (changes requested):
One small change requested to the Docker image tag, otherwise LGTM

Commit:
Update CircleCI Docker image tag

Updates the Docker image tag used in the `.circleci/config.yml` to
match the one used in the `docker-compose.yml`.
